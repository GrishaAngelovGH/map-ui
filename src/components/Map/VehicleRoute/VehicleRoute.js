import { Fragment } from 'react'

import L from 'leaflet'
import { Marker, Polyline, useMap } from 'react-leaflet'
import MarkerIcon from '../MarkerIcon'

const polyline = [
  [51.50917249679021, -0.12439680208002753],
  [51.51045456127713, -0.12160730480114733],
  [51.51108222550184, -0.11978340271756788],
  [51.51191687202163, -0.11902165540212936],
  [51.51294513550582, -0.11803460254261373],
  [51.51329038279241, -0.11633525726622364],
  [51.51303665952414, -0.11490832218710867],
  [51.513497366524774, -0.1131166066178557],
  [51.513864593410354, -0.11092792392206086]
]

const limeOptions = { color: 'lime', weight: 10 }

const VehicleRoute = () => {
  const map = useMap()
  map.fitBounds(L.polyline(polyline).getBounds())

  return (
    <Fragment>
      <Polyline pathOptions={limeOptions} positions={polyline} />
      {
        polyline.map((v, i) => (
          <Marker key={i} position={v} icon={MarkerIcon} />
        ))
      }
    </Fragment>
  )
}

export default VehicleRoute