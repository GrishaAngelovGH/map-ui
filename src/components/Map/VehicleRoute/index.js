import VehicleRoute from './VehicleRoute'

export default VehicleRoute