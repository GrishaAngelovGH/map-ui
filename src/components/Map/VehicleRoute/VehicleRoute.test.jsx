import { render } from '@testing-library/react'
import { vi } from 'vitest'

import { MapContainer } from 'react-leaflet'
import VehicleRoute from './VehicleRoute'

vi.mock('react-leaflet', async () => {
  const originalModule = await vi.importActual('react-leaflet')

  return {
    __esModule: true,
    ...originalModule,
    // Named export mocks
    Polyline: vi.fn(),
    MapContainer: originalModule.MapContainer
  }
})

test('should render VehicleRoute component', () => {
  const view = render(
    <MapContainer minZoom={10} maxZoom={16}>
      <VehicleRoute />
    </MapContainer>
  )

  expect(view).toMatchSnapshot()
})