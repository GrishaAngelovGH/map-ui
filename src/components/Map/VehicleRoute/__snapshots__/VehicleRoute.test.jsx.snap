// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render VehicleRoute component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="left: 0px; top: 0px;"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          />
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          />
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          >
            <div
              class="leaflet-control-zoom leaflet-bar leaflet-control"
            >
              <a
                aria-disabled="false"
                aria-label="Zoom in"
                class="leaflet-control-zoom-in"
                href="#"
                role="button"
                title="Zoom in"
              >
                <span
                  aria-hidden="true"
                >
                  +
                </span>
              </a>
              <a
                aria-disabled="true"
                aria-label="Zoom out"
                class="leaflet-control-zoom-out leaflet-disabled"
                href="#"
                role="button"
                title="Zoom out"
              >
                <span
                  aria-hidden="true"
                >
                  −
                </span>
              </a>
            </div>
          </div>
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          >
            <div
              class="leaflet-control-attribution leaflet-control"
            >
              <a
                href="https://leafletjs.com"
                title="A JavaScript library for interactive maps"
              >
                <svg
                  aria-hidden="true"
                  class="leaflet-attribution-flag"
                  height="8"
                  viewBox="0 0 12 8"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h12v4H0z"
                    fill="#4C7BE1"
                  />
                  <path
                    d="M0 4h12v3H0z"
                    fill="#FFD500"
                  />
                  <path
                    d="M0 7h12v1H0z"
                    fill="#E0BC00"
                  />
                </svg>
                 Leaflet
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
      style="position: relative;"
      tabindex="0"
    >
      <div
        class="leaflet-pane leaflet-map-pane"
        style="left: 0px; top: 0px;"
      >
        <div
          class="leaflet-pane leaflet-tile-pane"
        />
        <div
          class="leaflet-pane leaflet-overlay-pane"
        />
        <div
          class="leaflet-pane leaflet-shadow-pane"
        />
        <div
          class="leaflet-pane leaflet-marker-pane"
        />
        <div
          class="leaflet-pane leaflet-tooltip-pane"
        />
        <div
          class="leaflet-pane leaflet-popup-pane"
        />
      </div>
      <div
        class="leaflet-control-container"
      >
        <div
          class="leaflet-top leaflet-left"
        >
          <div
            class="leaflet-control-zoom leaflet-bar leaflet-control"
          >
            <a
              aria-disabled="false"
              aria-label="Zoom in"
              class="leaflet-control-zoom-in"
              href="#"
              role="button"
              title="Zoom in"
            >
              <span
                aria-hidden="true"
              >
                +
              </span>
            </a>
            <a
              aria-disabled="true"
              aria-label="Zoom out"
              class="leaflet-control-zoom-out leaflet-disabled"
              href="#"
              role="button"
              title="Zoom out"
            >
              <span
                aria-hidden="true"
              >
                −
              </span>
            </a>
          </div>
        </div>
        <div
          class="leaflet-top leaflet-right"
        />
        <div
          class="leaflet-bottom leaflet-left"
        />
        <div
          class="leaflet-bottom leaflet-right"
        >
          <div
            class="leaflet-control-attribution leaflet-control"
          >
            <a
              href="https://leafletjs.com"
              title="A JavaScript library for interactive maps"
            >
              <svg
                aria-hidden="true"
                class="leaflet-attribution-flag"
                height="8"
                viewBox="0 0 12 8"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h12v4H0z"
                  fill="#4C7BE1"
                />
                <path
                  d="M0 4h12v3H0z"
                  fill="#FFD500"
                />
                <path
                  d="M0 7h12v1H0z"
                  fill="#E0BC00"
                />
              </svg>
               Leaflet
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
